modal/_container_entrypoint.py

```diff
@@ -494,9 +501,36 @@ def import_function(function_def: api_pb2.Function, ser_cls, ser_fun) -> Importe
         cls, fun = load_function_from_module(module, function_def.function_name)
 
     # The decorator is typically in global scope, but may have been applied independently
+    active_stub = None
     if isinstance(fun, (FunctionHandle, AioFunctionHandle)):
         _function_proxy = synchronizer._translate_in(fun)
         fun = _function_proxy.get_raw_f()
+        active_stub = _function_proxy._stub
+    elif module is not None:
+        # This branch is reached in the special case that the imported function is 1) not serialized, and 2) isn't a FunctionHandle - i.e, not decorated at definition time
+        # Look for stubs in the same module as the function
+        # This is not necessarily enough, and the active stub might not even be loaded, but it's a best effort
+        # TODO(elias): Use some kind of global automatic "Stub registry" instead of looking at stubs in global scope of the module?
```

Comment:
yeah i think that would be a lot cleaner!

Reply:
Done!